File: LeetCode 75 Study Plan to Ace Interviews/Q206 - Reverse Linked List/solution.java
// solution 1: iterative using while loop
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        if (head == null) {
            return head;
        }
        ListNode tail = head;
        while (tail.next != null) {
            ListNode formerHead = head;
            head = tail.next;
            tail.next = tail.next.next;
            head.next = formerHead;
        }
        return head;
    }
}

File: LeetCode 75 Study Plan to Ace Interviews/Q206 - Reverse Linked List/solution.py
# solution 1: iterative using while loop
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head == None:
            return head
        tail = head
        while tail.next != None:
            formerHead = head
            head = tail.next
            tail.next = tail.next.next
            head.next = formerHead
        return head
